addons/mail/static/src/js/systray.js

```diff
@@ -136,16 +121,162 @@ var MessagingMenu = Widget.extend({
+    },
+
+    // Private
+
+    /**
+     * Make RPC and get current user's activity details
+     * @private
+     */
+    _getActivityData: function(){
+        var self = this;
+
+        return self._rpc({
+            model: 'res.users',
+            method: 'get_current_user_activities',
+        }).then(function (data) {
+            self.isCounterUpdated = false;
+            self.activities = data;
+            self.activityCounter = _.reduce(data, function(total_count, p_data){ return total_count + p_data.total_count; }, 0);
+            self.$('.o_notification_counter').text(self.activityCounter);
+            self.$el.toggleClass('o_no_notification', !self.activityCounter);
+        });
+    },
+    /**
+     * Get particular model view to redirect on click of activity scheduled on that model.
+     * @private
+     * @param {string} model
+     */
+    _getActivityModelViewID: function (model) {
+        return this._rpc({
+            model: model,
+            method: 'get_activity_view_id'
+        });
+    },
+    /**
+     * Check weather activity systray dropdown open or not
+     * @private
+     */
+    _isOpen: function () {
+        return this.$el.hasClass('open');
+    },
+    /**
+     * Update(render) activity system tray view on activity updation.
+     * @private
+     */
+    _updateActivityPreview: function () {
+        var self = this;
+        self._getActivityData().then(function (){
+            self.$activities_preview.html(QWeb.render('mail.chat.ActivityMenuPreview', {
+                activities : self.activities
+            }));
+        });
+    },
+    /**
+     * update counter based on activity status(created or Done)
+     * @private
+     * @param {Object} [data] key, value to decide activity created or deleted
+     * @param {String} [data.type] notification type
+     * @param {Boolean} [data.activity_deleted] when activity deleted
+     * @param {Boolean} [data.activity_created] when activity created
+     */
+    _updateCounter: function (data) {
+        if (data) {
+            if (data.activity_created) {
+                this.activityCounter ++;
+            }
+            if (data.activity_deleted && this.activityCounter > 0) {
+                this.activityCounter --;
+            }
+            this.isCounterUpdated = true;
+            this.$('.o_notification_counter').text(this.activityCounter);
+            this.$el.toggleClass('o_no_notification', !this.activityCounter);
         }
     },
+
+
+    // Handlers
+
+    /**
+     * Redirect to particular model view
+     * @private
+     * @param {MouseEvent} event
+     */
+    _onActivityFilterClick: function (event) {
+        event.stopPropagation();
+        var $target = $(event.currentTarget);
+        var context = {};
+        context['search_default_activities_' + $target.data('filter')] = 1;
+        this._getActivityModelViewID($target.data('res_model')).then(function(data) {
+            web_client.do_action({
```

Comment:
I do not think that you should use the web_client here. you should simply do this.do_action